pkg/lwip/contrib/lwip.c

```diff
@@ -179,94 +230,106 @@ void lwip_bootstrap(void)
 {
     (void)_netif_add;   /* in case it is not used */
     /* TODO: do for every eligible netdev */
-#ifdef LWIP_NETIF_NUMOF
+#ifdef LWIP_NETIF_USED
```

Comment:
As you define `LWIP_NETIF_NUMOF_...` if the module is not there as 0 and then `LWIP_NETIF_NUMOF` as the sum of those you could instead of the `#undef; #define` mess above just remove `LWIP_NETIF_USED` altogether and use

```suggestion
#if IS_ACTIVE(LWIP_NETIF_NUMOF)
```

or

```suggestion
#if LWIP_NETIF_NUMOF > 0
```

(as it is always defined) here. I think if you use C conditionals here
```suggestion
    if (LWIP_NETIF_NUMOF > 0) { // ...
```

we can even remove most of the pre-processor mess in this file (the array definitions might pose a problem though).

Reply:
> (the array definitions might pose a problem though).

Might be worth trying out putting them as `static` variables in function the function.

Reply:
First, I tried it in that way `#if LWIP_NETIF_NUMOF > 0` but it does not work for `LWIP_NETIF_NUMOF_*` definitions with `ARRAY_SIZE` definitions. Since the preprocessor doesn't know anything about types, the `sizeof` operator is not allowed in conditionals.

I will try `IS_ACTIVE(LWIP_NETIF_NUMOF)`.

Reply:
Defining the static variables inside an `if`-block with the `IS_USED` macro should work, for example
```c
if (IS_USED(MODULE_NETDEV_TAP)) {
    static struct netif netif[LWIP_NETIF_NUMOF];
    for (unsigned i = 0; i < LWIP_NETIF_NUMOF_NETDEV_TAP; i++) {
        netdev_tap_setup(&netdev_taps[i], &netdev_tap_params[i]);
        if (_netif_add(&netif[_netif++], &netdev_taps[i], lwip_netdev_init,
                       tcpip_input) == NULL) {
            DEBUG("Could not add netdev_tap device\n");
            return;
        }
    }
}

But would it would produce changes for the whole file because of the indentation. I didn't wanted to change to much with this PR. If you think that it makes sense, I can change it.

Reply:
As this file is not very thought-through to begin with, let's do this now that we make this new step ;-). An alternative, after sleeping on it, could be to split the devices up file-wise like we do with GNRC.

Reply:
Unfortunately, we have to use the preprocessor conditional `#if IS_USED(MODULE_NETDEV_TAP)`, with C conditional the code is compiled and `netdev_tap_t` is unknown type :worried:

Reply:
Yes we have to remove the `#ifdef` around the include then of course as well!

Reply:
I pushed a commit that places all static variables inside the function. Wherever possible, it replaces preprocessor conditions with C conditions.